core/templates/domain/exploration/ExplorationDraftObjectFactory.ts

```diff
@@ -20,24 +20,75 @@
 import { downgradeInjectable } from '@angular/upgrade/static';
 import { Injectable } from '@angular/core';
 
+export interface IAddStateNameChangeList {
+  'cmd': 'add_state';
+  'state_name': string;
+}
+
+export interface IDeleteStateChangeList {
+  'cmd': 'delete_state';
+  'state_name': string;
+}
+
+export interface IEditExplorationPropertyChangeList {
+  'cmd': 'edit_exploration_property';
+  'new_value': Object;
+  'old_value': Object;
+  'property_name': string;
+}
+
+export interface IEditStatePropertyChangeList {
+  'cmd': 'edit_state_property';
+  'new_value': Object;
+  'old_value': Object;
+  'property_name': string;
+  'state_name': string;
+}
+
+export interface IRenameStateChangeList {
+  'cmd': 'rename_state';
+  'new_state_name': string;
+  'old_state_name': string;
+}
+
+export interface IRevertChangeList {
+  'cmd': 'AUTO_revert_version_number';
```

Comment:
Why AUTO is in uppercase?

Reply:
No idea! This is the command that was already there.

Reply:
Ah, okay.